### PR TITLE
qa_crowbarsetup.sh: Create /etc/crowbar/repos.yml for Cloud 6

### PR DESCRIPTION
**Pull request by vuntz**
For Cloud 6 (after M6), we don't specify custom repositories in the
provisioner barclamp, but in /etc/crowbar/repos.yml.

This is needed for the changes in https://github.com/crowbar/crowbar-core/pull/102

### CODE REVIEW

**Comment by vuntz** [comment]
cc @MaximilianMeister

File: scripts/qa_crowbarsetup.sh [hunk 1]
```diff
@@ -1113,6 +1113,54 @@ function do_set_repos_skip_checks()
...
+            echo "  $repo_name:"
+            echo "    url: '$repo_url'"
+        fi
+    done
+}
+
+function create_repos_yml()
+{
+    local repos_yml="/etc/crowbar/repos.yml"
+    local tmp_yml="${repos_yml}.tmp"
```
Comment (MaximilianMeister) [inline scripts/qa_crowbarsetup.sh]: we could use `mktemp` here, like we do in some other place IIRC

File: scripts/qa_crowbarsetup.sh [hunk 2]
```diff
@@ -1113,6 +1113,54 @@ function do_set_repos_skip_checks()
 }
 
 
+function create_repos_yml_for_platform()
+{
+    local platform=$1
+    local tftpboot_dir=$2
+    shift; shift
+    local platform_created=
```
Comment (MaximilianMeister) [inline scripts/qa_crowbarsetup.sh]: is the `=` in the end intended here?

Comment (vuntz) [inline scripts/qa_crowbarsetup.sh]: fixed

File: scripts/qa_crowbarsetup.sh (hunk 1 above)
Comment (vuntz) [inline scripts/qa_crowbarsetup.sh]: fixed

**Comment by MaximilianMeister** [comment]
haven't tested but code looks good
:+1:

**Comment by bmwiedemann** [comment]
code looks good. My only concern is that we might want to add some nicer way for this for users and rather test that.

**Comment by vuntz** [comment]
@bmwiedemann at the moment, there's no nicer way except for yast (see https://github.com/yast/yast-crowbar/pull/35). But I agree that if/once we add some API for this, we should use the API.

**Comment by jdsn** [comment]
:+1:  lgtm

**Comment by vuntz** [comment]
Just checked in the current gate job: file was created correctly, and provisioner proposal didn't have the autoyast repo. It also deploys openstack, so I think we're good.

**Pull request merged**